oauth-common/src/main/java/io/strimzi/kafka/oauth/validator/OAuthIntrospectionValidator.java

```diff
@@ -272,14 +272,14 @@ public TokenInfo validate(String token) {
 
         } catch (IOException e) {
             addIntrospectHttpMetricErrorTime(e, requestStartTime);
-            throw new RuntimeException("Failed to introspect token - send, fetch or parse failed: ", e);
+            throw new ValidationException("Failed to introspect token - send, fetch or parse failed: ", e);
         }
 
         boolean active;
         try {
             active = response.get("active").asBoolean();
         } catch (Exception e) {
-            throw new RuntimeException("Failed to introspect token - invalid response: \"active\" attribute is missing or not a boolean (" + response.get("active") + ")");
+            throw new ValidationException("Failed to introspect token - invalid response: \"active\" attribute is missing or not a boolean (" + response.get("active") + ")");
```

Comment:
Looking at the Javadocs `get()` and `asBoolean()` don't seem to throw but instead return `null` and `false`. Should we check the return value of `get()` instead?

Reply:
You mean, if the JSON is:
```
{
    "active": "True"
}
```
the code will currently actually set `active` to `false`, because only string with case-sensitive value of "true" will be converted to the boolean value of `true`.

That's a good point. While authorization servers should return a boolean attribute we should maybe be more strict, and we currently don't really detect the 'not a boolean' case.

Reply:
Fixed.